### PR TITLE
Add LCM, IPA, and Scene Lora in Video Inference.

### PR DESCRIPTION
**Pull request by bubbliiiing**
1. LCM
New LCM acceleration added. When producing images of 512x512 resolution, totaling 16 frames, with all models preheated, the generation time is approximately 100 seconds.
image

2. IPA
Added IPA control that can improve likeness. After usage, the similarity of portraits is mostly above 0.65.
image

3. Scene Lora
Added a new scene, Lora, which is used in the same way as text2photo.
image

### CODE REVIEW

File: scripts/easyphoto_infer.py [hunk 1]
```diff
@@ -576,10 +576,6 @@ def easyphoto_infer_forward(
     if lcm_accelerate:
         lcm_lora_name_and_weight = "lcm_lora_sdxl:0.40" if sdxl_pipeline_flag else "lcm_lora_sd15:0.80"
 
-    # get random seed
```
Comment (wuziheng) [inline scripts/easyphoto_infer.py:579]: 为什么remove -1 的seed ，没有看到原因

File: scripts/easyphoto_infer.py [hunk 2]
```diff
@@ -1730,20 +1728,76 @@ def easyphoto_video_infer_forward(
...
             passed_userid_list.append(idx)
+        else:
+            last_user_id_none_num = 0
 
-    if len(user_ids) == len(passed_userid_list):
+    if len(user_ids) == last_user_id_none_num:
         return "Please choose a user id.", None, None, []
 
+    # check the version of controlnets
+    controlnet_version = get_controlnet_version()
```
Comment (wuziheng) [inline scripts/easyphoto_infer.py:1759]: duplicate with https://github.com/aigc-apps/sd-webui-EasyPhoto/blob/0260f834d2431dc0416c86c363de79379df1dd4a/scripts/easyphoto_infer.py#L533 ~ L580

File: scripts/easyphoto_ui.py [hunk 3]
```diff
@@ -1261,10 +1286,38 @@ def update_uuids(_num_of_faceid):
...
                                         display_score = gr.Checkbox(label="Display Face Similarity Scores", value=False)
-                                        makeup_transfer = gr.Checkbox(label="Video MakeUp Transfer", value=False)
+                                        ipa_control = gr.Checkbox(label="IP-Adapter Control", value=False)
                                         face_shape_match = gr.Checkbox(label="Video Face Shape Match", value=False)
                                     with gr.Row():
+                                        makeup_transfer = gr.Checkbox(label="Video MakeUp Transfer", value=False)
                                         video_interpolation = gr.Checkbox(label="Video Interpolation", value=False)
+                                        lcm_accelerate = gr.Checkbox(label="LCM Accelerate", value=False)
+
+                                        def lcm_change(lcm_accelerate):
```
Comment (wuziheng) [inline scripts/easyphoto_ui.py:1296]: https://github.com/aigc-apps/sd-webui-EasyPhoto/blob/0260f834d2431dc0416c86c363de79379df1dd4a/scripts/easyphoto_ui.py#L792

是不是遇到这种都必须要重新原地define一个。当然我觉得原地define 也没啥问题，好修改，就是有点冗长

**Review by wuziheng — COMMENTED** [review]
annimate diff update should be review carefully

**Comment by wuziheng** [comment]
commit aff26f0a0 has bug 
截屏2023-12-11 下午7 26 19

should fix  @bubbliiiing

File: scripts/easyphoto_infer.py (hunk 1 above)
Comment (bubbliiiing) [reply scripts/easyphoto_infer.py:579]: 换位置了

Comment (bubbliiiing) [reply scripts/easyphoto_infer.py:579]: 另一个pr忘了删掉原来的
image

**Comment by bubbliiiing** [comment]
> commit [aff26f0](https://github.com/aigc-apps/sd-webui-EasyPhoto/commit/aff26f0a08cbb778c25551ae521fe93c1c9f762d) has bug 截屏2023-12-11 下午7 26 19
> 
> should fix @bubbliiiing

已经修复。
这是因为在scripts/easyphoto_utils/animatediff/animatediff_infv2v.py的125行，获取了control.control_model.uncond_image_emb[context]，但uncond_image_emb事实上长度只有1，并没有拓展到可以获取的长度；
而与之对应的control.control_model.image_emb的长度是生成视频帧数的2倍，使用context可以正常获取。

如图为controlnet中的定义，具体为何为1还需debug，因为看起来在controlnet定义中，正负两个的长度应该一样：
image

**Review by wuziheng — APPROVED** [review]
LGTM

**Comment by wuziheng** [comment]
768*512 16frame + superresolution on A10
W/O LCM ： 250s
W LCM ：155s

**Pull request merged**